src/routes/articles/update-article.e2e.test.js

```diff
@@ -0,0 +1,199 @@
+			const expectedError = {
+				"error": {
+					"message": "article must be a string"
+				},
+				"status": "failed"
+			}
+			return fixtures.api()
+				.patch(`/api/v1/articles/${article.id}`)
+				.set('Authorization', `Bearer ${accessToken}`)
+				.send({ 
+					title : updatedInfo.title,
+					image : updatedInfo.image,
+					article : 123,
+					published : updatedInfo.published
+				})
+				.expect(400, expectedError)
+		})
+		it('should return 400 if image is not a valid url',
+		 async () => {
+				const expectedError = {
+					"error": {
+						"message": "image must be a string"
+					},
+					"status": "failed"
+				}
+				return fixtures.api()
+					.patch(`/api/v1/articles/${article.id}`)
+					.set('Authorization', `Bearer ${accessToken}`)
+					.send({
+						title : updatedInfo.title,
+						image : 123,
+						article : updatedInfo.content,
+						published : updatedInfo.published
+					})
+					.expect(400, expectedError)
+			})
+		it('should return 400 if published is not a boolean',
+			async () => {
+				const expectedError = {
+					"error": {
+						"message": "published must be a boolean"
+					},
+					"status": "failed"
+				}
+				return fixtures.api()
+					.patch(`/api/v1/articles/${article.id}`)
+					.set('Authorization', `Bearer ${accessToken}`)
+					.send({
+						title : updatedInfo.title,
+						image : updatedInfo.image,
+						article : updatedInfo.content,
+						published : 123
+					})
+					.expect(400, expectedError)
+			})
+
+		it('should return 404 if article is not found', async () => 
+			
+			fixtures.api()
+				.patch(`/api/v1/articles/${article.id + 2}`)
+				.set('Authorization', `Bearer ${accessToken}`)
+				.send({
+					title : updatedInfo.title,
+					image : updatedInfo.image,
+					article : updatedInfo.content,
+					published : updatedInfo.published
+					 })
+				.expect(404)
+				.then((res) => {
+					expect(res.body.status).eql('fail')
+					expect(res.body.message).eql('Article does not exist')
+				})
+		)
+
+	})
+	describe('Success', () => {
+		it('should return 200 if article is updated', async () => 
+			fixtures.api()
+				.patch(`/api/v1/articles/${article.id}`)
+				.set('Authorization', `Bearer ${accessToken}`)
+				.send({ 
+					title : updatedInfo.title,
+					image : updatedInfo.image,
+					article : updatedInfo.content,
+					published : updatedInfo.published
+				})
+				.expect(200)
+
+		)
+		it('should return the right response', async () =>
+			fixtures.api()
+				.patch(`/api/v1/articles/${article.id}`)
+				.set('Authorization', `Bearer ${accessToken}`)
+				.send({ 
+					title : updatedInfo.title,
+					image : updatedInfo.image,
+					article : updatedInfo.content,
+					published : updatedInfo.published
+				})
+				.expect(200)
+				.then((res) => {
+					expect(res.body.data.userId).to.be.an('number')
+					expect(res.body.data.articleId).to.be.an('number')
+					expect(res.body.data.createdOn).to.be.a('string')
```

Comment:
yes please, i need clarification

Reply:
okay, 

so in the other api test, we were testing creation of an article, with that there are autogenerated fields, namely the `id` of the article and the `createdOn` of the article, because there's no way of knowing their exact values to be able to compare, we just make do with ensuring their types are at least correct. That is,to check  if the `id` is a number and so on

for the update article api test, we have already inserted the article and therefore we have an idea what the `id` of the article, we already know the `userId` and the `createdOn` so we don't have to settle for just their types

you're already doing that in the `expect` below these ones so that's perfect

Reply:
oh okay, i  understand it perfectly now